web/src/lib/components/StallView.svelte

```diff
@@ -1,49 +1,106 @@
 <script lang="ts">
-    import PublicAuctionCard from "../components/PublicAuctionCard.svelte";
-    import type { User } from "../types/user";
-    import { type ILoader, getEntities, getStall, getProfile, ErrorHandler } from "../services/api";
+    import AuctionCard from "$lib/components/AuctionCard.svelte";
+    import type { IAccount, User } from "../types/user";
+    import { type ILoader, getEntities, getStall, getProfile, ErrorHandler, postEntity, putEntity } from "../services/api";
     import { onMount } from "svelte";
     import { token, user } from "../stores";
     import StallNotFound from "./StallNotFound.svelte";
     import { afterNavigate } from '$app/navigation';
     import type { IEntity } from '$lib/types/base';
+    import Loading from "$lib/components/Loading.svelte";
+    import Avatar from "$lib/components/Avatar.svelte";
+    import { goto } from '$app/navigation';
 
-    export let stallName;
+    export let title;
+    export let card;
+    export let editor;
+    export let stallOwnerNym;
+    export let showNewButton: boolean = true;
     export let loader: ILoader;
     export let entities: IEntity[] | null = null;
 
+    export let newEntity: () => IEntity;
+    export let onView: (entity: IEntity) => void = (_) => { };
+    export let onCreated: () => void = () => { };
+
     let stallOwner: User;
+    let storeAccount: IAccount;
     let currentTab = "ACTIVE AUCTIONS";
+    let tabList = ['ACTIVE AUCTIONS', 'PAST AUCTIONS']
     let loading = true;
+    let currentEntity: IEntity | undefined;
 
     function fetchProfile(tokenValue) {
         getProfile(tokenValue,
             u => {
                 user.set(u);
+                // if user and no stallNym -> go to own stall
+                if (stallOwnerNym === "") {
+                    goto(`/stall/${u.nym}`);
+                }
             });
     }
 
-    function fetchStall(stallName: string) {
-        getStall(stallName, 
+    function fetchStall(stallOwnerNym: string) {
+        getStall(stallOwnerNym, 
             s => {
                 stallOwner = s;
```

Comment:
Just name the parameter stallOwner rather than s and that's it ..

Reply:
still needed